crates/biome_js_analyze/src/lint/nursery/no_secrets.rs

```diff
@@ -0,0 +1,254 @@
+
+        // Since regex matching is expensive, we run in threads
+        // - Added a mutex for "found", i.e. I want to exit early if any
+        // pattern matches or string contains is found
+        // - Currently hardcoded to 4 threads. Can be adjusted later.
+        // - I do realize that it might be overkill for smaller strings.
+        // Maybe can check that in future.
+        let handles: Vec<_> = SENSITIVE_PATTERNS
+            .chunks(patterns_per_thread)
+            .map(|chunk| {
+                let text = Arc::clone(&text);
+                let result = Arc::clone(&result);
+                let found = Arc::clone(&found);
+                thread::spawn(move || {
+                    for (pattern, comment) in chunk {
+                        if *found.lock().unwrap() {
+                            return;
+                        }
+                        let matched = match pattern {
+                            Pattern::Regex(re) => re.is_match(&text),
+                            Pattern::Contains(substring) => text.contains(substring),
+                        };
+                        if matched {
+                            let mut guard = result.lock().unwrap();
+                            if guard.is_none() {
+                                *guard = Some(*comment);
+                                *found.lock().unwrap() = true;
+                            }
+                            return;
+                        }
+                    }
+                })
+            })
+            .collect();
+
+        for handle in handles {
+            handle.join().unwrap();
+        }
+
+        let guard = result.lock().unwrap();
+        if let Some(comment) = *guard {
+            return Some(comment);
+        }
+
+        if is_high_entropy(&text) {
+            Some("High entropy value detected")
+        } else {
+            None
+        }
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                node.range(),
+                markup! { "Potential secret found." },
+            )
+            .note(markup! { {state} }),
+        )
+    }
+}
+
+fn is_high_entropy(text: &str) -> bool {
+    let entropy = calculate_shannon_entropy(text);
+    entropy > 4.5 // TODO: Make this optional, or controllable
+}
+
+/**
+ * Inspired by https://github.com/nickdeis/eslint-plugin-no-secrets/blob/master/utils.js#L93
+ * Adapted from https://docs.rs/entropy/latest/src/entropy/lib.rs.html#14-33
+ * Calculates Shannon entropy to measure the randomness of data. High entropy values indicate potentially
+ * secret or sensitive information, as such data is typically more random and less predictable than regular text.
+ * Useful for detecting API keys, passwords, and other secrets within code or configuration files.
+ * @param {*} str
+ */
```

Comment:
nit: We generally use `///` syntax for doc comments.

Reply:
Done!